src/cli/outdated_command.zig

```diff
@@ -180,92 +186,282 @@ pub const OutdatedCommand = struct {
+
+        var workspace_pkg_ids: std.ArrayListUnmanaged(PackageID) = .{};
+        for (pkg_resolutions, 0..) |resolution, pkg_id| {
+            if (resolution.tag != .workspace and resolution.tag != .root) continue;
+            try workspace_pkg_ids.append(allocator, @intCast(pkg_id));
+        }
+
+        const converted_filters = converted_filters: {
+            const buf = try allocator.alloc(struct { []const u32, bool }, filters.len);
+            for (filters, buf) |filter, *converted| {
+                const is_path = filter.len > 0 and filter[0] == '.';
+
+                const joined_filter = if (is_path)
+                    strings.withoutTrailingSlash(path.joinAbsString(original_cwd, &[_]string{filter}, .posix))
+                else
+                    filter;
+
+                if (joined_filter.len == 0) {
+                    converted.* = .{ &.{}, is_path };
+                    continue;
+                }
 
-        try switch (Output.enable_ansi_colors) {
-            inline else => |enable_ansi_colors| printOutdatedInfoTable(manager, root_pkg_deps, enable_ansi_colors),
+                const length = bun.simdutf.length.utf32.from.utf8.le(joined_filter);
+                const convert_buf = try allocator.alloc(u32, length);
+
+                const convert_result = bun.simdutf.convert.utf8.to.utf32.with_errors.le(joined_filter, convert_buf);
+                if (!convert_result.isSuccessful()) {
+                    // nothing would match
+                    converted.* = .{ &.{}, false };
+                    continue;
+                }
+
+                converted.* = .{ convert_buf[0..convert_result.count], is_path };
+            }
+            break :converted_filters buf;
         };
+        defer {
+            for (converted_filters) |converted| {
+                const filter, _ = converted;
+                allocator.free(filter);
+            }
+            allocator.free(converted_filters);
+        }
+
+        // move all matched workspaces to front of array
+        var i: usize = 0;
+        while (i < workspace_pkg_ids.items.len) {
+            const workspace_pkg_id = workspace_pkg_ids.items[i];
+
+            const matched = matched: {
+                for (converted_filters) |converted| {
+                    const filter, const is_path_filter = converted;
+
+                    if (is_path_filter) {
+                        if (filter.len == 0) continue;
+                        const res = pkg_resolutions[workspace_pkg_id];
+
+                        const res_path = switch (res.tag) {
+                            .workspace => res.value.workspace.slice(string_buf),
+                            .root => FileSystem.instance.top_level_dir,
+                            else => unreachable,
+                        };
+
+                        const abs_res_path = path.joinAbsString(FileSystem.instance.top_level_dir, &[_]string{res_path}, .posix);
```

Comment:
probably should use joinAbsStringBuf incase something else is making use of the shared string buffer?